config/routes.rb

```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get "/users", to: "users#index" #  requires a view
+  get '/users/:user_id', to: 'users#show' #  requires a view
+  get '/users/:user_id/posts', to: 'posts#index'
+  get '/users/:user_id/posts/:id', to: 'posts#show'
```

Comment:
- [ ]  [OPTIONAL] You can use the following code snippet to represent exactly the routes you wrote instead of hard coding the routes.
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```